src/gf.c

```diff
@@ -1930,6 +1929,54 @@ jl_value_t *jl_gf_invoke(jl_tupletype_t *types0, jl_value_t **args, size_t nargs
     return jl_call_method_internal(mfunc, args, nargs);
 }
 
+JL_DLLEXPORT jl_value_t *jl_get_invoke_lambda(jl_methtable_t *mt,
+                                              jl_typemap_entry_t *entry,
+                                              jl_tupletype_t *tt)
+{
+    if (!jl_is_leaf_type((jl_value_t*)tt))
+        return jl_nothing;
```

Comment:
> overly strict

Yeah. There are currently very feel users and most of them should have leaf type (forwarding) so I didn't bother handling more complex cases.

> I guess you're mimicking `jl_get_specialization1`

Right.

> You also need to check that none of the elements are types of types like that other method too then.

Added. I assume the comments below (inexact and caching) does not apply anymore then?